### PR TITLE
Add AWS and GCP Fabric Cloud Router connections

### PR DESCRIPTION
**Pull request by manu-equinix**
Adds the below Fabric cloud router(FCR) connections types and related tests and examples

- FCR to AWS
- FCR to GCP

### CODE REVIEW

File: tests/connection_e2e_fcr2aws_test.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCloudRouter2AwsCreateConnection(t *testing.T) {
+	// retryable errors in terraform testing.
+	terraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+		TerraformDir: "../examples/fabric-cloud-router/fg2aws",
```
Comment (thogarty) [inline tests/connection_e2e_fcr2aws_test.go:13]: Update path to "../examples/fabric/v4/cloudRouterConnectivity/cloudRouter2aws"

File: tests/connection_e2e_fcr2aws_test.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	})
+
+	defer terraform.Destroy(t, terraformOptions)
+
+	terraform.InitAndApply(t, terraformOptions)
+	output := terraform.Output(t, terraformOptions, "fg2aws_connection_result")
```
Comment (thogarty) [inline tests/connection_e2e_fcr2aws_test.go:19]: Output needs to match the output name in the main.tf of the terraform it is referencing. In this case the output is "connection_result"

File: tests/connection_e2e_fcr2gcp_test.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCloudRouter2GcpCreateConnection(t *testing.T) {
+	// retryable errors in terraform testing.
+	terraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+		TerraformDir: "../examples/fabric-cloud-router/fcr2gcp",
```
Comment (thogarty) [inline tests/connection_e2e_fcr2gcp_test.go:13]: Update path to "../examples/fabric/v4/cloudRouterConnectivity/cloudRouter2gcp"

File: tests/connection_e2e_fcr2gcp_test.go [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+	})
+
+	defer terraform.Destroy(t, terraformOptions)
+
+	terraform.InitAndApply(t, terraformOptions)
+	output := terraform.Output(t, terraformOptions, "fcr2gcp_connection_result")
```
Comment (thogarty) [inline tests/connection_e2e_fcr2gcp_test.go:19]: Output is named "connection_result" in main.tf of example script. Needs to match in order to pass the test after run.

File: tests/connection_e2e_fcr2port_test.go [hunk 5]
```diff
@@ -16,6 +16,6 @@ func TestCloudRouter2PortCreateConnection(t *testing.T) {
 	defer terraform.Destroy(t, terraformOptions)
 
 	terraform.InitAndApply(t, terraformOptions)
-	output := terraform.Output(t, terraformOptions, "fcr2port_result")
+	output := terraform.Output(t, terraformOptions, "fcr2port_connection_result")
```
Comment (thogarty) [inline tests/connection_e2e_fcr2port_test.go:19]: This should remain "fcr2port_result" to match the output in the corresponding script main.tf file.

File: tests/connection_e2e_fcr_test.go [hunk 6]
```diff
@@ -10,12 +10,21 @@ import (
 func TestCloudRouterCreate(t *testing.T) {
 	// retryable errors in terraform testing.
 	terraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
-		TerraformDir: "../examples/fabric/v4/cloudRouterConnectivity/cloudRouter",
+		TerraformDir: "../examples/fabric/v4/cloudRouter",
```
Comment (thogarty) [inline tests/connection_e2e_fcr_test.go:13]: This test is running the same directory twice. Just needs to remove the redundant code for a single test on cloudRouter.

**Review by thogarty — CHANGES_REQUESTED** [review]
Left some comments, but the rest looks good to me!

File: tests/connection_e2e_fcr2aws_test.go (hunk 2 above)
Comment (manu-equinix) [reply tests/connection_e2e_fcr2aws_test.go]: Ack.

File: tests/connection_e2e_fcr2gcp_test.go (hunk 3 above)
Comment (manu-equinix) [reply tests/connection_e2e_fcr2gcp_test.go]: Ack.

File: tests/connection_e2e_fcr2gcp_test.go (hunk 4 above)
Comment (manu-equinix) [reply tests/connection_e2e_fcr2gcp_test.go]: Ack. I have made all the files consistent and updated to a generic "connection_result" in all places.

File: tests/connection_e2e_fcr2port_test.go (hunk 5 above)
Comment (manu-equinix) [reply tests/connection_e2e_fcr2port_test.go]: Ack.

File: tests/connection_e2e_fcr_test.go (hunk 6 above)
Comment (manu-equinix) [reply tests/connection_e2e_fcr_test.go]: Ack.

File: tests/connection_e2e_fcr2gcp_test.go [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCloudRouter2GcpCreateConnection(t *testing.T) {
+	// retryable errors in terraform testing.
+	terraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+		TerraformDir: ".../examples/fabric/v4/cloudRouterConnectivity/cloudRouter2gcp",
```
Comment (thogarty) [inline tests/connection_e2e_fcr2gcp_test.go:13]: Sorry to nit, there's 3`.` instead of two here. You've got my approval for the PR though.

**Comment by manu-equinix** [comment]
Updated code to address all review comments. Please check and approve @ctreatma @thogarty

**Review by thogarty — DISMISSED** [review]
Left one comment about the 3`.`s in the path, but the rest of it looks great. Approval given.

File: tests/connection_e2e_fcr2gcp_test.go (hunk 7 above)
Comment (manu-equinix) [reply tests/connection_e2e_fcr2gcp_test.go:13]: Good catch! Updated

**Comment by manu-equinix** [comment]
> Left one comment about the 3`.`s in the path, but the rest of it looks great. Approval given.

fixed

**Review by ctreatma — APPROVED** [review]

**Pull request merged**